sdk/identity/identity/src/credentials/deviceCodeCredential.ts

```diff
@@ -123,12 +126,22 @@ export class DeviceCodeCredential implements TokenCredential {
       scopes: scopeArray
     };
 
-    logger.info("Sending devicecode request");
+    logger.info(`DeviceCodeCredential invoked. Scopes: ${scopeArray.join(", ")}`);
 
-    return this.msalClient.acquireTokenFromCache(scopeArray).catch((e) => {
+    try {
+      return this.msalClient.acquireTokenFromCache(scopeArray);
+    } catch (e) {
       if (e instanceof AuthenticationRequired) {
         try {
-          return this.acquireTokenByDeviceCode(deviceCodeRequest, scopeArray);
+          const token = await this.acquireTokenByDeviceCode(deviceCodeRequest, scopeArray);
+          logger.info(
+            formatSuccess(
+              `DeviceCodeCredential successfully retrieved the access token. Scopes: ${scopeArray.join(
```

Comment:
Are other credentials, sending out an equally detailed message for success and failure logs ?

Reply:
This is something I want to investigate! I can do that in this PR or in a separate PR.